### PR TITLE
Only one real listener per `listen` call

### LINKED ISSUE #2534: Polymer events creates a new bound handler for every `listen` call
As discussed in person, this could be made slightly more optimal by looking to see if a listener has been created already.

Fixing this would allow `listen` to be called n times with the same arguments, while ensuring that only one listener is ever actually added.

### PR DESCRIPTION
**Pull request by azakus**
Actually use event listener cache correctly

Fixes #2534 Polymer Gestures creates a new bound handler for every `listen` call

### CODE REVIEW

**Comment by samccone** [comment]
Is there a test we can put in place for this?

**Comment by azakus** [comment]
@samccone yep

**Comment by cdata** [comment]
A good test might be..
1. Add a listener twice
2. Verify that the event is only handled once
3. Remove the listener
4. Verify that the event is no longer handled

File: src/standard/events.html [hunk 1]
```diff
@@ -80,8 +80,12 @@
      * @param {string} methodName Name of handler method on `this` to call.
      */
     listen: function(node, eventName, methodName) {
-      this._listen(node, eventName,
-        this._createEventHandler(node, eventName, methodName));
+      var handler = this._recallEventHandler(this, eventName, node, methodName);
+      // reuse cache'd handler
```
Comment (cdata) [inline src/standard/events.html]: :beers:

**Comment by cdata** [comment]
Relevant for https://github.com/PolymerElements/iron-selector/pull/64

**Comment by azakus** [comment]
@sorvell or @kevinpschaaf PTAL

File: test/unit/events.html [hunk 2]
```diff
@@ -141,6 +141,41 @@
           assert.deepEqual(el._removed[3], {target: 'div', event: 'bar'});
         });
       });
+
+      suite('Event Listener Cache', function() {
```
Comment (samccone) [inline test/unit/events.html]: woot

**Comment by sorvell** [comment]
LGTM

**Pull request merged**